java/sage/Carny.java

```diff
@@ -1818,11 +2325,993 @@ public void windowClosing(java.awt.event.WindowEvent evt)
+
+          // This only happens for active favorites which should number less than 500, so this isn't
+          // going to amount to anything crazy for the GC. We are trying to take advantage of any
+          // scratch space left in the currently allocated cache list.
+          cache.airWorkCache.setOffset();
+          // Also check the rem airs so we're sure we get EVERYTHING in the DB that applies to this
+          // Favorite included in the group.
+          //
+          // We are passing true to ignoreDisabledFlag because we have already filtered out the
+          // disabled agents making this additional check a waste of time.
+          currAgent.getRelatedAirings(remAirs, controlCPUUsage, true, legacyKeyword, remAirsMap,
+            cache, cache.sbCache);
+
+          int remWorkCacheSize = cache.airWorkCache.size;
+          callback.appendCapacityNewLoveAirSet(remWorkCacheSize);
+          for (int i = cache.airWorkCache.offset; i < remWorkCacheSize; i++)
+            callback.addNewLoveAirSet(cache.airWorkCache.data[i]);
+
+          // Remove the entries we just appended and gets rid of the offset so they don't break
+          // anything below.
+          cache.airWorkCache.clearOffset();
+        }
+
+        // Check to see if this Agent is a Favorite who has a keep at most limit set with
+        // manual deleting. In that case, we don't add any of its future airings to the
+        // mustSeeSet or to the pots, but we do put them in the loveAirSet.
+        boolean dontScheduleThisAgent = false;
+        if (isFavorite && currAgent.testAgentFlag(Agent.DONT_AUTODELETE_FLAG) &&
+          currAgent.getAgentFlag(Agent.KEEP_AT_MOST_MASK) > 0)
+        {
+          int fileCount = 0;
+          for (int j = 0, potsWorkCacheSize = cache.airWorkCache.size; j < potsWorkCacheSize; j++)
+          {
+            MediaFile mf = wiz.getFileForAiring(cache.airWorkCache.data[j]);
+            if (mf != null && mf.isCompleteRecording())
+              fileCount++;
+          }
+          int keepAtMost = currAgent.getAgentFlag(Agent.KEEP_AT_MOST_MASK);
+          if (fileCount >= keepAtMost)
+          {
+            dontScheduleThisAgent = true;
+          }
+        }
+
+        int potsWorkCacheSize = cache.airWorkCache.size;
+        boolean negator = currAgent.isNegativeNelly();
+        if (negator)
+          callback.appendCapacityBlackBalled(potsWorkCacheSize);
+        else
+          callback.appendCapacityAirSet(potsWorkCacheSize);
+
+        for (int j = 0; j < potsWorkCacheSize; j++)
+        {
+          Airing agentPot = cache.airWorkCache.data[j];
+          if (negator)
+          {
+            callback.addBlackBalled(agentPot);
+            continue;
+          }
+
+          // Skip this stuff
+          Show show = agentPot.getShow();
+          if ((show != null && show.title.equalsIgnoreCase(paidProgRez)) || wiz.isNoShow(agentPot.showID))
+            continue;
+
+          callback.addAirSet(agentPot);
```

Comment:
Doesn't this seem like it's pointless? I had this in my old code...but when reviewing it, it seemed unnecessary since this agent actually came from that set. You should be able to add some logging to determine if this ever does actually add it.